packages/wallet/src/redux/protocols/virtual-funding/reducer.ts

```diff
@@ -301,49 +301,80 @@ function waitForGuarantorFundingReducer(
   action: WalletAction,
 ) {
   const { processId, protocolLocator } = protocolState;
-  if (routesToIndirectFunding(action, protocolLocator)) {
-    const result = indirectFunding.indirectFundingReducer(
-      protocolState.indirectGuarantorFunding,
+  if (
+    !routesToIndirectFunding(action, protocolLocator) &&
+    !routesToConsensusUpdate(action, protocolLocator)
+  ) {
+    console.warn(
+      `Expected indirectFunding or consensusUpdate action, received ${action.type} instead`,
+    );
+    return { protocolState, sharedData };
+  }
+  if (routesToConsensusUpdate(action, protocolLocator)) {
+    let indirectApplicationFunding: consensusUpdate.ConsensusUpdateState;
+    ({
+      protocolState: indirectApplicationFunding,
+      sharedData,
+    } = consensusUpdate.consensusUpdateReducer(
+      protocolState.indirectApplicationFunding,
       sharedData,
       action,
-    );
-    if (indirectFunding.isTerminal(result.protocolState)) {
-      switch (result.protocolState.type) {
-        case 'IndirectFunding.Success':
-          // Once funding is complete we allow consensusUpdate to send commitments
-          const applicationFundingResult = consensusUpdate.consensusUpdateReducer(
-            protocolState.indirectApplicationFunding,
-            result.sharedData,
-            clearedToSend({
-              processId,
-              protocolLocator: makeLocator(protocolLocator, CONSENSUS_UPDATE_PROTOCOL_LOCATOR),
-            }),
-          );
-          return {
-            protocolState: states.waitForApplicationFunding({
-              ...protocolState,
-              indirectApplicationFunding: applicationFundingResult.protocolState,
-            }),
-            sharedData: applicationFundingResult.sharedData,
-          };
-        case 'IndirectFunding.Failure':
-          throw new Error(`Indirect funding failed: ${result.protocolState.reason}`);
-
-        default:
-          return unreachable(result.protocolState);
-      }
-    } else {
-      return {
-        protocolState: states.waitForGuarantorFunding({
-          ...protocolState,
+    ));
+    switch (indirectApplicationFunding.type) {
+      // TODO: Properly handle the success case
+      // We don't expect this to ever happen now but we should future-proof it
+      case 'ConsensusUpdate.Success':
+      case 'ConsensusUpdate.Failure':
+        return {
+          protocolState: states.failure({
+            reason: 'Consensus Update failed or succeeded too early',
+          }),
+          sharedData,
+        };
+      default:
+        return { protocolState: { ...protocolState, indirectApplicationFunding }, sharedData };
+    }
+  }
+  const result = indirectFunding.indirectFundingReducer(
```

Comment:
Should this instead be
```
else if (!routesToIndirectFunding(action, protocolLocator)) { ... }
```
?

If you use some local enum like above, then the logic here is simpler.

Reply:
That should be handled by the initial check ```  !routesToIndirectFunding(action, protocolLocator) &&
    !routesToConsensusUpdate(action, protocolLocator)``` if I understand your comment correctly.